seal5/cli/build.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2023 TUM Department of Electrical and Computer Engineering.
+#
+# This file is part of Seal5.
+# See https://github.com/tum-ei-eda/seal5.git for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Command line subcommand for building Seal5 LLVM."""
+
+from seal5.flow import Seal5Flow
+
+
+def add_build_options(parser):
+    build_parser = parser.add_argument_group("build options")
+    build_parser.add_argument(
+        "-n",
+        "--name",
+        metavar="NAME",
+        nargs=1,
+        type=str,
+        default="default",
+        help="Environment name (default: %(default)s)",
+    )
+    build_parser.add_argument(
+        "-dir",
+        nargs="?",
+        type=str,
+        default="/var/tmp/seal5_demo/",
+        help="LLVM directory (default: %(default)s",
+    )
+    build_parser.add_argument(
+        "--target",
+        "-t",
+        default="all",
+        help="All things that should be build in the make process",
+    )
+    build_parser.add_argument(
+        "--config",
+        default="release",
+        help="Choose build Config in Settings.yml",
+    )
+    build_parser.add_argument(
+        "--verbose",
+        default=False,
+        action="store_true",
+        help="Verbose printing of steps into console",
+    )
+
+
+def get_parser(subparsers):
+    """ "Define and return a subparser for the build subcommand."""
+    parser = subparsers.add_parser("build", description="Build Seal5.")
+    parser.set_defaults(func=handle)
+    add_build_options(parser)
+    return parser
+
+
+def handle(args):
+    """Callback function which will be called to process the build subcommand"""
+    name = args.name[0] if isinstance(args.name, list) else args.name
```

Comment:
Passing the name via CMDline should only be required for `seal5 init`.

Can we make the name argument optional, if the name can be automatically retrieved from the `settings.yml`?

Alternatively, we could implement `Seal5Flow.from_directory()` which would handle this.

Reply:
Is there a link to the settings.yml file in the env? Or where should I know which settings.yml file should be fetched for this?

Reply:
You mean that I should create sth like a seal5_home environment var and integrate it into https://github.com/tum-ei-eda/seal5/blob/main/seal5/flow.py#L62 ?

Reply:
For now just assume that the settings file is located at `{dir}/.seal5/settings.yml`.

I realized that the Seal5Settings do not yet have a `name: str = "..."` attribute. This would need to be added to get the name for the Seal5Flow...

Reply:
@PhilippvK

Reply:
If we have a non-default location for the meta_dir, user might need to specify the path to the meta_dir instead and we can figure out the llvm dir location be checking `self.settings.directory`.

Reply:
> For now just assume that the settings file is located at `{dir}/.seal5/settings.yml`.
> 
> I realized that the Seal5Settings do not yet have a `name: str = "..."` attribute. This would need to be added to get the name for the Seal5Flow...

Well this was my first env that destroyed my original repo with git applies. Or do you mean {dest}/.seal5/settings.yml` ?

Reply:
I would prefer a Seal5_home variable (os.environ['Seal5_Home']) that is set with the init step and read in the following steps